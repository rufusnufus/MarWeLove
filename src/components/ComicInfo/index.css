.comicinfo {
  display: flex;
  color: white;
  margin-bottom: 130px;
}

.comicinfo__name {
  margin-top: 162px;
  margin-left: 236px;
  font-weight: bold;
  font-size: 50px;
  line-height: 74px;
}

.comicinfo__description {
  width: 474px;
  margin-left: 266px;
  margin-top: 20px;
  font-weight: bold;
  font-size: 25px;
  line-height: 74px;
}

.comicinfo__image {
  width: 463px;
  height: 463px;
  background-size: cover;
  margin-left: 100px;
  margin-top: 154px;
}
