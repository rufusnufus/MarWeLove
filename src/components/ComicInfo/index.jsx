import React from 'react'

import './index.css'

function ComicInfo({ data }) {
  return (
    <div className="comicinfo">
      <div className="comicinfo__text">
        <div className="comicinfo__name">{data.title}</div>
        <div className="comicinfo__description">{data.description}</div>
      </div>
      <div
        className="comicinfo__image"
        style={{ backgroundImage: `url(${data.thumbnail?.path}.${data.thumbnail?.extension})` }}
      />
    </div>
  )
}

export default ComicInfo
