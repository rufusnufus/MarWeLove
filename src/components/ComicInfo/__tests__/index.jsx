/**
 * @jest-environment jsdom
 */

import React from 'react'
import 'jest-styled-components'
import { cleanup, render } from '@testing-library/react'
import ComicInfo from '../index'

afterEach(cleanup)

const mockData = {
  id: 1003,
  title: 'Sentry, the (Trade Paperback)',
  description: 'A forgotten hero, he must unravel the conspiracy to erase his memory.',
  thumbnail: {
    path: '',
    extension: ''
  }
}

const mockDataWithoutThumbnail = {
  id: 1003,
  title: 'Sentry, the (Trade Paperback)',
  description: 'A forgotten hero, he must unravel the conspiracy to erase his memory.'
}

describe('CharacterInfo', () => {
  it('Character Name', () => {
    const { queryByText } = render(<ComicInfo data={mockData} />)

    expect(queryByText('Sentry, the (Trade Paperback)')).toBeTruthy()
  })

  it('Character Description', () => {
    const { queryByText } = render(<ComicInfo data={mockData} />)

    expect(queryByText('A forgotten hero, he must unravel the conspiracy to erase his memory.')).toBeTruthy()
  })

  it('Render with thumbnail', () => {
    const { container } = render(<ComicInfo data={mockData} />)

    expect(container).toMatchSnapshot()
  })

  it('Render without thumbnail', () => {
    const { container } = render(<ComicInfo data={mockDataWithoutThumbnail} />)

    expect(container).toMatchSnapshot()
  })
})

//   describe('Comic', () => {
//     it('Render', () => {
//       const { queryByText } = render(<Card type="comics" data={mockData} />)

//       expect(queryByText('comic')).toBeTruthy()
//     })
//   })
