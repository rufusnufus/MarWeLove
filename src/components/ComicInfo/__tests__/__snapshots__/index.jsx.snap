// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterInfo Render with thumbnail 1`] = `
<div>
  <div
    class="comicinfo"
  >
    <div
      class="comicinfo__text"
    >
      <div
        class="comicinfo__name"
      >
        Sentry, the (Trade Paperback)
      </div>
      <div
        class="comicinfo__description"
      >
        A forgotten hero, he must unravel the conspiracy to erase his memory.
      </div>
    </div>
    <div
      class="comicinfo__image"
      style="background-image: url(.);"
    />
  </div>
</div>
`;

exports[`CharacterInfo Render without thumbnail 1`] = `
<div>
  <div
    class="comicinfo"
  >
    <div
      class="comicinfo__text"
    >
      <div
        class="comicinfo__name"
      >
        Sentry, the (Trade Paperback)
      </div>
      <div
        class="comicinfo__description"
      >
        A forgotten hero, he must unravel the conspiracy to erase his memory.
      </div>
    </div>
    <div
      class="comicinfo__image"
      style="background-image: url(undefined.undefined);"
    />
  </div>
</div>
`;
